lib/fastlane/actions/install_carthage.rb

```diff
@@ -2,19 +2,67 @@ module Fastlane
   module Actions
     class CarthageAction < Action
       def self.run(params)
-        Actions.sh('carthage bootstrap')
+
+        cmd = ['carthage bootstrap']
+
+        cmd << '--use-ssh' if params[:use_ssh]
+        cmd << '--use-submodules' if params[:use_submodules]
+        cmd << '--no-use-binaries' if !params[:use_binaries]
+        cmd << '--platform' << params[:platform]
+
+        Actions.sh(cmd.join(' '))
       end
 
       def self.description
         "Runs `carthage bootstrap` for your project"
       end
 
-      def self.author
-        "bassrock"
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :use_ssh,
+                                       env_name: "FL_CARTHAGE_USE_SSH",
+                                       description: "Use SSH for downloading GitHub repositories",
+                                       default_value: false,
+                                       is_string: false,
+                                       optional: true,
+                                       verify_block: Proc.new do |value|
+                                         raise "Please pass a valid use_ssh. Use one of the following: true, false" unless value.is_a?(TrueClass) || value.is_a?(FalseClass)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :use_submodules,
+                                       env_name: "FL_CARTHAGE_USE_SUBMODULES",
+                                       description: "Add dependencies as Git submodules",
+                                       default_value: false,
+                                       is_string: false,
+                                       optional: true,
+                                       verify_block: Proc.new do |value|
+                                         raise "Please pass a valid use_submodules. Use one of the following: true, false" unless value.is_a?(TrueClass) || value.is_a?(FalseClass)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :use_binaries,
+                                       env_name: "FL_CARTHAGE_NO_USE_BINARIES",
```

Comment:
`:use_binaries` == `FL_CARTHAGE_NO_USE_BINARIES`? 
Also, default_value is true so it will always use submodules?